web/src/app/user/UserComponent.js

```diff
@@ -1,18 +1,213 @@
+import 'react-bootstrap-table-next/dist/react-bootstrap-table2.min.css';
+import 'react-bootstrap-table2-paginator/dist/react-bootstrap-table2-paginator.min.css';
+import { Button, Col, Grid, Image, Row, Tabs, Tab } from 'react-bootstrap';
+import cx from 'classnames';
 import React, { Component } from 'react';
+import BootstrapTable from 'react-bootstrap-table-next';
+import BountyDetails from '../profile/BountyDetails/BountyDetails';
+import paginationFactory from 'react-bootstrap-table2-paginator';
 import PropTypes from 'prop-types';
+import { OPEN, COMPLETED, FAILED } from '../BountyState';
+import profileStyles from '../profile/ProfileStyles.scss';
+import styles from './UserComponentStyles.scss';
+import tableStyles from '../tableStyles.scss';
+
+const pageSize = 10;
+const tabArray = [OPEN, COMPLETED, FAILED]; // should be in the order of the tabs
 
 class UserComponent extends Component {
-  render() {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      tabKey: 1,
+      curPage: 1,
+      bountyFilter: OPEN,
+      curBounty: null
+    };
+  }
+
+  componentDidMount() {
+    this.refreshList();
+  }
+
+  tableChange = (type, newState) => {
+    if (type === 'pagination') {
+      this.setState({
+        curPage: newState.page
+      }, () => {
+        this.props.getPublicBounties(newState.page - 1, newState.sizePerPage, this.props.publicUser.displayName, this.state.bountyFilter);
+      });
+    }
+  }
+
+  refreshList() {
+    this.props.getPublicBounties(0, pageSize, this.props.publicUser.displayName, this.state.bountyFilter);
+  }
+
+  handleSelect = (key) => {
```

Comment:
Seems like this method is handling a bounty state filter change. Maybe renaming to handleBountyFilterChange or onBountyFilterChange could help with clarity